main.go

```diff
@@ -300,7 +302,10 @@ func filterData(pump pumps.Pump, keys []interface{}) []interface{} {
 	shouldTrim := SystemConfig.MaxRecordSize != 0 || pump.GetMaxRecordSize() != 0
 	filters := pump.GetFilters()
 	ignoreFields := pump.GetIgnoreFields()
-	if !filters.HasFilter() && !pump.GetOmitDetailedRecording() && !shouldTrim && len(ignoreFields) == 0 {
+	getDecodingResponse := pump.GetDecodedResponse()
+	getDecodingRequest := pump.GetDecodedRequest()
+	if !getDecodingRequest && !getDecodingResponse && !filters.HasFilter() && !pump.GetOmitDetailedRecording() && !shouldTrim && len(ignoreFields) == 0 {
```

Comment:
also, could you explain the logic in this conditional? it can be an inline comment or something, as it's a pretty long conditional

Reply:
Yes done - my addition to this code was `!getDecodingRequest `and `!getDecodingResponse` to check if the configs for both of those was false. Essentially the whole conditional is to see if all config options are empty or false.